pkgs/racket-index/scribblings/main/license.scrbl

```diff
@@ -1,75 +1,336 @@
 #lang scribble/doc
-@(require scribble/manual "private/utils.rkt")
+@(require scribble/manual
+          racket/cmdline
+          racket/match
+          "private/utils.rkt")
 
-@(define (copyright . strs) (apply verbatim #:indent 2 strs))
+@(define mode
+   (let ([mode 'docs])
+     (command-line
+      #:argv (for/list ([arg (in-vector (current-command-line-arguments))]
+                        ;; hack: with `raco setup`, might be e.g. #("setup")
+                        #:when (member arg '("--repo-root-dir" "--repo-src-dir")))
+               arg)
+      #:usage-help
+      "Use with “scribble ++arg” to generate license files for the source repository,"
+      "rather than the license page for the installed documentation."
+      #:once-any
+      [("--repo-root-dir") "Generate “LICENSE.txt” for the Git repository root."
+       (set! mode 'root)]
+      [("--repo-src-dir") "Generate “LICENSE.txt” for the “racket/src” directory."
+       (set! mode 'src)]
+      #:args ()
+      mode)))
 
-@main-page['license]
+@(define-syntax-rule (if-repo for-repo for-docs)
+   ;; For text that should be different in a LICENSE.txt file
+   ;; in the source tree than in installed documentation.
+   (if (eq? 'docs mode)
+       for-docs
+       for-repo))
+@(define-syntax-rule (when-repo arg ...)
+   (if-repo (list arg ...) null))
+@(define-syntax-rule (unless-repo arg ...)
+   (if-repo null (list arg ...)))
 
-Racket is distributed under the MIT license and the Apache version 2.0
-license, at your option. However, the Racket runtime system includes
+@(define (src-filepath . args)
+   (match mode
+     ['root
+      (apply filepath "racket/src/" args)]
+     ['src
+      (apply filepath args)]
+     ['doc
+      (error 'src-filepath "not supported in 'doc mode")]))
+
+@(define (tech #:doc doc #:indirect? indirect? . args)
+   ;; FIXME: make this work in Scribble.
+   ;; See https://github.com/racket/scribble/pull/337
+   (match args
+     ['("byte converter")
+      (apply seclink
+             #:indirect? #t
```

Comment:
Should `#t` here be `indirect?`.

Reply:
It is definitely just meant to be temporary, but that would make this hack slightly less ugly. (IIUC `indirect?` will always have to be `#t`, though, because `racket-index` shouldn't depend on `racket-doc`.)